overlord/snapstate/handlers.go

```diff
@@ -560,18 +560,20 @@ func (m *SnapManager) installPrereqs(t *state.Task, base string, prereq map[stri
 		}
 	}
 
-	// on systems without core or snapd need to install snapd to
-	// make interfaces work - LP: 1819318
+	// On classic systems that are already seeded, automatically
+	// install snapd snap (covers LP: 1819318). Not allowed for
+	// Ubuntu Core systems - requires remodeling.
 	var tsSnapd *state.TaskSet
 	snapdSnapInstalled, err := isInstalled(st, "snapd")
 	if err != nil {
 		return err
 	}
-	coreSnapInstalled, err := isInstalled(st, "core")
-	if err != nil {
-		return err
-	}
-	if base != "core" && !snapdSnapInstalled && !coreSnapInstalled {
+
+	// consider the state of seeding to avoid seed conflict error
+	deviceCtx, err := DevicePastSeeding(st, nil)
+	seeded := err == nil && deviceCtx != nil
```

Comment:
we usually just get "seeded" from state to check for this

Reply:
Changed this, also send question over MM for further consideration.